io/io/v7/src/TFile.cxx

```diff
@@ -97,7 +97,7 @@ class TV6Storage: public ROOT::Experimental::Internal::TFileStorageInterface {
   }
 
   void WriteMemoryWithType(std::string_view name, const void* address, TClass* cl) final {
-    fOldFile->WriteObjectAny(address, cl, name.to_string().c_str());
+     fOldFile->WriteObjectAny(address, cl, name.data());
```

Comment:
> Starting C++11 std::string .data() and .c_str() both provide null-terminated result

I somehow did not catch that.  Thanks!

> Could you point me to the comment in the code which states that it's required? 

Humm ... WriteObjectAny takes a const char* and use it as null terminated string, I am not sure why there would be a 'special' comment in this routine to make that explicit ...

It is clear that std::string_view::data is not appropriate here, as this returns the first characters of the view, which could for example be a sub-part of a larger string.